exercises/bank-account/canonical-data.json

```
@@ -0,0 +1,375 @@
+    },
+    {
+      "uuid": "0722d404-6116-4f92-ba3b-da7f88f1669c",
+      "description": "Reopened account does not retain balance",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": 50
+          },
+          {
+            "operation": "close"
+          },
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "balance"
+          }
+        ]
+      },
+      "expected": 0
+    },
+    {
+      "uuid": "ec42245f-9361-4341-8231-a22e8d19c52f",
+      "description": "Cannot withdraw more than deposited",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": 25
+          },
+          {
+            "operation": "withdraw",
+            "amount": 50
+          }
+        ]
+      },
+      "expected": { "error": "amount must be less than balance" }
+    },
+    {
+      "uuid": "4f381ef8-10ef-4507-8e1d-0631ecc8ee72",
+      "description": "Cannot withdraw negative",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": 100
+          },
+          {
+            "operation": "withdraw",
+            "amount": -50
+          }
+        ]
+      },
+      "expected": { "error": "amount must be greater than 0" }
+    },
+    {
+      "uuid": "d45df9ea-1db0-47f3-b18c-d365db49d938",
+      "description": "Cannot deposit negative",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": -50
+          }
+        ]
+      },
+      "expected": { "error": "amount must be greater than 0" }
+    },
+    {
+      "uuid": "ba0c1e0b-0f00-416f-8097-a7dfc97871ff",
+      "description": "Can handle concurrent transactions",
+      "comments": [
+        "This test is a good candidate to practice concurrency, which",
+        "may not be available or possible in the implementing language track.",
+        "Concurrency testing is often made by running the same operations in",
+        "parallel, and then checking that the final state is as expected."
+      ],
+      "scenarios": ["concurrent"],
+      "property": "bankAccount",
+      "input": {
+        "concurrent_operations": [
+          {
+            "operation": "deposit",
+            "amount": 1
+          },
+          {
+            "operation": "withdraw",
+            "amount": 1
+          }
+        ],
+        "operations": [
+          {
+            "operation": "balance"
+          }
```

Let's give the current operations some number (number of threads/fibers/etc)
```suggestion
      "input": {
        "initial_operations": [
          {
            "operation": "open"
          }
        ],
        "concurrent_operations": {
          "number": 1000,
          "operations": [
            {
              "operation": "deposit",
              "amount": 1
            },
            {
              "operation": "withdraw",
              "amount": 1
            }
          ]
        },
        "finishing_operations": [
          {
            "operation": "balance"
          }
```